packages/vite-extensions/vite-website-cache-paths.ts

```diff
@@ -0,0 +1,93 @@
+import { readdirSync, statSync, writeFileSync } from 'node:fs';
+import { join } from 'node:path';
+
+/**
+ * We want to cache all files from the website root directory, and the assets directory, except for the files listed below.
+ */
+const patternsToNotCache = [
+	/**
+	 * Static files that are not needed for the website to function offline.
+	 * TODO: Check if these are needed in the production build.
+	 */
+	'/package.json',
+	'/README.md',
+	'/.DS_Store',
+	'/index.cjs',
+	'/index.d.ts',
+	/\/lib\/.*/, // Remote lib files
+	/\/test-fixtures\/.*/, // Test fixtures
+	/**
+	 * WordPress remote assets that are fetched on demand by Playground.
+	 */
+	/^\/wp-[\w+((\.\d+)?)]+\/.*/,
+	/**
+	 * Demos are not needed for the website to function offline.
+	 */
+	/^\/demos\/.*/,
+	/**
+	 * Files needed only by the Playground.WordPress.Net server.
+	 */
+	'/.htaccess',
+	/\/.*\.php$/, // PHP files used by the website server
+	/**
+	 * WordPress, PHP, and SQLite files that are loaded during boot.
+	 * By loading these files during boot, we can ensure that only the current PHP/WordPress/SQLite files are cached.
+	 * These files are large, so loading only the current files will reduce the bandwidth usage.
+	 */
+	/^\/assets\/php_.*\.wasm$/, // PHP WASM files
+	/^\/assets\/php_.*\.js$/, // PHP JS files
+	/^\/assets\/wp-.*\.zip$/, // WordPress zip files
+	/^\/assets\/sqlite-database-integration-[\w]+\.zip/, // SQLite plugin
+];
+
+/**
+ * For Playground to work offline we need to cache all the files that are needed for the website to function.
+ * Playground uses this list at the end of the boot process to cache files.
+ */
+export const websiteCachePathsPlugin = () => {
```

Comment:
Let's use a more descriptive name, e.g. `listAssetsRequiredForOfflineModePlugin()`. It would also be nice to accept the filename as an argument so that the person reading `website/vite.config.ts` can immediately understand what this is about, e.g.:

```ts
			listAssetsRequiredForOfflineModePlugin({
				outputFile: path('./assets-required-for-offline-mode.json');
			}),
```